File: client/java/BigFoxJavaClient/src/main/java/vn/com/onesoft/bigfox/io/message/core/sc/SCValidationCode.java
package vn.com.onesoft.bigfox.io.message.core.sc;

/*
 * Author: HuongNS
 * Copyright @ 2015 by OneSoft.,JSC
 * 
 */


import io.netty.channel.Channel;
import vn.com.onesoft.bigfox.io.core.session.ConnectionManager;
import vn.com.onesoft.bigfox.io.message.annotations.Message;
import vn.com.onesoft.bigfox.io.message.annotations.Property;
import vn.com.onesoft.bigfox.io.message.base.MessageIn;
import vn.com.onesoft.bigfox.io.message.core.cs.CSClientInfo;
import vn.com.onesoft.bigfox.io.message.core.tags.CoreTags;

/**
 *
 * @author HuongNS
 */
@Message(tag = CoreTags.SC_VALIDATION_CODE, name = "SC_VALIDATION_CODE", isCore = true)
public class SCValidationCode extends MessageIn {

    @Property(name = "validationCode")
    private int validationCode;

    @Override
    public void execute(Channel channel) {
        ConnectionManager.getInstance().setValidationCode(validationCode);
        ConnectionManager.getInstance().write(new CSClientInfo());
    }

}

File: client/ios/BigFoxClient/BigFoxClient/message/core/sc/SCValidationCode.m
//
//  SCValidationCode.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/29/15.
//  Copyright © 2015 QuyenNX. All rights reserved.
//

#import "SCValidationCode.h"
#import "ConnectionManager.h"
#import "CSClientInfo.h"
#import "CoreTags.h"

static const int TAG = SC_VALIDATION_CODE;
@implementation SCValidationCode
@synthesize validationCode;

- (id) init {
    self = [super init:true];
    if (self) {
        self.tag = TAG;
    }
    return self;
}

-(void) execute {
    [[ConnectionManager getInstance] setValidationCode: self.validationCode];
    CSClientInfo* info = [[CSClientInfo alloc] init];
    [[ConnectionManager getInstance] write: info];
}
@end
